Update backtesting tutorial (#591)
langchain-ai/langsmith-docs: docs/evaluation/tutorials/backtesting.mdx

# Backtesting

Deploying your application is just the beginning of a continuous improvement process.

> Typically, you'll want to refine your system by enhancing prompts, language models, tools, and architectures.

Reviewer: ```suggestion
After you deploy to production, you'll want to refine your system by enhancing prompts, language models, tools, and architectures.
```

> While the initial datasets you created before deployment are useful for testing new versions, you now have access to real historical data. This data provides a more accurate representation of the inputs your application will encounter in the future.

Reviewer: ```suggestion
```

> Backtesting involves assessing new versions of your application using this historical data and comparing the new outputs to the original ones.

Reviewer: ```suggestion
Backtesting involves assessing new versions of your application using historical data and comparing the new outputs to the original ones.
```

> This approach can offer a clearer indication of whether the new version is an improvement over the current deployment compared to evaluations using pre-production datasets.

Reviewer: ```suggestion
Compared to evaluations using pre-production datasets, backtesting offers a clearer indication of whether the new version of your application is an improvement over the current deployment. 
```

Here are the basic steps for backtesting:

1. Select sample runs from your production tracing project to test against.
2. Transform the run inputs into a dataset and record the run outputs as an initial experiment against that dataset.
3. Execute your new system on the new dataset and compare the results of the experiments.

This process will provide you with a new dataset of representative inputs, which you can version and use for backtesting your models.

:::info Ground truth data
Often, you won't have definitive "ground truth" answers available.
In such cases, you can manually label the outputs or use evaluators that don't rely on reference data.
If your application allows for capturing ground-truth labels, for example by allowing users to leave feedback, we strongly recommend doing so.
:::

> 

Reviewer: Add a sentence here about what the tutorial covers. eg In this tutorial, we will go over XYZ..

## Setup

### Configure the environment

> Install and set environment variables. This guide requires `langsmith>=0.2.4`.

Reviewer: `This guide requires `langsmith>=0.2.4`.` does this only apply if users are self-hosting?

Reviewer 2: no its for all, we use some sdk methods in the code snippets that are only available in that version
